extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/rule/RuleHelper.scala

```diff
@@ -47,6 +47,6 @@ trait RuleHelper extends Rule[LogicalPlan] {
     plan.withNewChildren(newChildren)
   }
 
-  def ugi: UserGroupInformation = AuthZUtils.getAuthzUgi(spark.sparkContext)
```

Comment:
This change may affect test cases: https://github.com/apache/kyuubi/blob/cad5a392f3195d8f3b16c293a11ac54c2f090871/extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/IcebergCatalogRangerSparkExtensionSuite.scala#L68-L87

@pan3793 Is it possible to have multiple ugi in the spark engine?

Reply:
Different sparkSession may have different authz ugi, I will revert this change first.